core/metrics-core-service/src/main/java/org/hawkular/metrics/core/dropwizard/HawkularMetricsRegistryListener.java

```diff
@@ -154,4 +275,15 @@ private void createMetrics(String metricName, List<Metric<?>> metrics) {
                 );
     }
 
+    private void deleteMetrics(String metricName, List<MetricId<?>> metrics) {
```

Comment:
Why ``List<MetricId<?>>`` and not ``Observable<MetricId<?>>`` ? The above onCounterRemoved etc should just instead of creating a List create an Observable.